skorch/callbacks/training.py

```diff
@@ -752,10 +752,18 @@ def initialize(self):
             **self._f_kwargs()
         )
         self.checkpoint_.initialize()
+        return self
 
     def on_train_end(self, net, **kwargs):
         self.checkpoint_.save_model(net)
         self.checkpoint_._sink("Final checkpoint triggered", net.verbose)
+        return self
 
     def __getattr__(self, attr):
+        if attr == 'checkpoint_':
+            # this should only happen when unpickling
+            try:
+                return object.__getattribute__(self, attr)
+            except AttributeError:
+                return None
```

Comment:
This `__getattr__` was introduced in https://github.com/skorch-dev/skorch/pull/536 and that PR was to fix the interaction between `LoadInitState` and `TrainEndCheckpoint`.

I think I prefer removing `__getattr__` and have `LoadInitState` handle `TrainEndCheckpoint` directly: #778